src/Sulu/Bundle/MediaBundle/Collection/Manager/CollectionManager.php

```diff
@@ -582,7 +587,46 @@ public function delete($id)
         $collectionTitle = $collectionMeta ? $collectionMeta->getTitle() : null;
         $locale = $collectionMeta ? $collectionMeta->getLocale() : null;
 
+        if (!$forceRemoveChildren) {
+            $user = $this->getCurrentUser();
+
+            if (null !== $user) {
+                $totalChildResourcesWithoutPermissions = $this->collectionRepository->countUnauthorizedChildCollectionsOfRootCollection(
+                    $id,
+                    $user,
+                    $this->permissions[PermissionTypes::DELETE]
+                );
+
+                if ($totalChildResourcesWithoutPermissions) {
+                    $childCollectionResourcesWithoutPermissions = $this->collectionRepository->findUnauthorizedChildCollectionResourcesOfRootCollection(
+                        $id,
+                        $user,
+                        $this->permissions[PermissionTypes::DELETE],
+                        3
+                    );
+
+                    throw new InsufficientChildPermissionsException(
+                        $childCollectionResourcesWithoutPermissions,
+                        $totalChildResourcesWithoutPermissions,
+                        PermissionTypes::DELETE
+                    );
+                }
+            }
+
+            $totalChildResources = $this->countAllChildrenByRootCategory($id);
```

Comment:
I think we don't need this count as we could at the end just do `count($childResources)` to return a total so we don't need this additonal method on the repositories to just count something and trigger similar queries against the database.